providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -101,10 +100,21 @@ const char *dh_gen_type_id2name(int id)
     return NULL;
 }
 
-static int dh_gen_type_name2id(const char *name)
+static int dh_gen_type_name2id(const char *name, int type)
 {
     size_t i;
 
+    if (strcmp(name, "default") == 0) {
+#ifdef FIPS_MODULE
+        if (type == DH_FLAG_TYPE_DHX)
+            return DH_PARAMGEN_TYPE_FIPS_186_4;
+#else
+        if (type == DH_FLAG_TYPE_DHX)
+            return DH_PARAMGEN_TYPE_FIPS_186_2;
+#endif
+        return DH_PARAMGEN_TYPE_GENERATOR;
```

Comment:
Why the inconsistency of using "default" generator type with the pre-initialized default value in case the pkey type is DH? The pre-initialized for FIPS provider is DH_PARAMGEN_TYPE_FIPS_186_4 and for "default" generator parameter it is DH_PARAMGEN_TYPE_GENERATOR.

Reply:
Oops. That was a mistake. Fixed.